src/energyplus/Test/HeatExchangerDesiccantBalancedFlow_GTest.cpp

```diff
@@ -0,0 +1,92 @@
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include <gtest/gtest.h>
+#include "EnergyPlusFixture.hpp"
+
+#include "../ForwardTranslator.hpp"
+
+#include "../../model/Model.hpp"
+#include "../../model/HeatExchangerDesiccantBalancedFlow.hpp"
+#include "../../model/HeatExchangerDesiccantBalancedFlow_Impl.hpp"
+#include "../../model/HeatExchangerDesiccantBalancedFlowPerformanceDataType1.hpp"
+#include "../../model/HeatExchangerDesiccantBalancedFlowPerformanceDataType1_Impl.hpp"
+
+#include "../../model/AirLoopHVAC.hpp"
+#include "../../model/AirLoopHVAC_Impl.hpp"
+#include "../../model/Node.hpp"
+#include "../../model/Node_Impl.hpp"
+#include "../../model/AirLoopHVACOutdoorAirSystem.hpp"
+#include "../../model/AirLoopHVACOutdoorAirSystem_Impl.hpp"
+#include "../../model/HVACTemplates.hpp"
+
+#include <utilities/idd/HeatExchanger_Desiccant_BalancedFlow_FieldEnums.hxx>
+#include <utilities/idd/IddEnums.hxx>
+
+using namespace openstudio::energyplus;
+using namespace openstudio::model;
+using namespace openstudio;
+
+TEST_F(EnergyPlusFixture, ForwardTranslator_HeatExchangerDesiccantBalancedFlow) {
+  Model m;
+
+  AirLoopHVAC loop = addSystemType3(m).cast<AirLoopHVAC>();
+
+  AirLoopHVACOutdoorAirSystem oaSystem = loop.airLoopHVACOutdoorAirSystem().get();
+
+  Node oaNode = oaSystem.outboardOANode().get();
+
+  HeatExchangerDesiccantBalancedFlow hx(m);
+  EXPECT_TRUE(hx.addToNode(oaNode));
+
+  ForwardTranslator ft;
+  Workspace w = ft.translateModel(m);
+
+  WorkspaceObjectVector idfHXs(w.getObjectsByType(IddObjectType::HeatExchanger_Desiccant_BalancedFlow));
+  ASSERT_EQ(1u, idfHXs.size());
+  WorkspaceObject idfHX(idfHXs[0]);
+
+  boost::optional<WorkspaceObject> woAvailabilitySchedule(idfHX.getTarget(HeatExchanger_Desiccant_BalancedFlowFields::AvailabilityScheduleName));
+  EXPECT_TRUE(woAvailabilitySchedule);
+  if (woAvailabilitySchedule) {
+    EXPECT_EQ(woAvailabilitySchedule->iddObject().type(), IddObjectType::Schedule_Constant);
+  }
+  EXPECT_NE("", idfHX.getString(HeatExchanger_Desiccant_BalancedFlowFields::RegenerationAirInletNodeName).get());
+  EXPECT_NE("", idfHX.getString(HeatExchanger_Desiccant_BalancedFlowFields::RegenerationAirOutletNodeName).get());
+  EXPECT_NE("", idfHX.getString(HeatExchanger_Desiccant_BalancedFlowFields::ProcessAirInletNodeName).get());
+  EXPECT_NE("", idfHX.getString(HeatExchanger_Desiccant_BalancedFlowFields::ProcessAirOutletNodeName).get());
```

Comment:
`EXPECT_EQ(hx.primaryAirOutletModelObject()->nameString(), idfHX.getString(HeatExchanger_Desiccant_BalancedFlowFields::RegenerationAirInletNodeName).get())` would be better.

Here you won't know whether you made a mistake mapping a node to the wrong E+ field.